lib/config/fileconf.go

```diff
@@ -875,11 +875,11 @@ func (a *App) CheckAndSetDefaults() error {
 	if a.URI == "" {
 		return trace.BadParameter("missing URI")
 	}
-	// Check if the application name contains spaces. Don't allow spaces because
-	// for trusted clusters the name is used to construct the vanity domain that
-	// the application will be running under.
-	if ok := strings.Contains(a.Name, " "); ok {
-		return trace.BadParameter("application name %q can not contain spaces as it may be used as part of the application domain", a.Name)
+	// Check if the application name is a valid subdomain. Don't allow names that
+	// are invalid subdomains because for trusted clusters the name is used to
+	// construct the domain that the application will be available at.
+	if ok := utils.ValidSubdomain(a.Name); !ok {
+		return trace.BadParameter("application name %q must be a valid DNS subdomain", a.Name)
```

Comment:
I'd definitely leave a reference to the docs where the DNS subdomain format has been specified or described if this is a user-facing error - e.g. at least https://tools.ietf.org/html/rfc1034#section-3.5

Reply:
I've added a link, it's not valid yet, but will be valid very soon.